src/include/sof/common.h

```diff
@@ -9,6 +9,12 @@
 #ifndef __SOF_COMMON_H__
 #define __SOF_COMMON_H__
 
+/*
+ * The most basic alignment macro with no checks, suitable for C, assembly and
+ * for linker scripts
+ */
+#define ALIGN_UP_INTERNAL(val, align) (((val) + (align) - 1) & ~((align) - 1))
```

Comment:
> > The most basic alignment macro with no checks, suitable for C, assembly and for linker scripts
> 
> This comment is still misleading:
> 
>     * maybe you meant "suitable for C _static initializers_"?
>     * it seems to be contradicting this other comment below:

@marc-hb I tried to improve it, please, check.

> > memory.h values are _also_ copied unprocessed to the .x[.in] linker script
> 
> that's because ALIGN is unprocessed while ALIGN_UP_INTERNAL is. Too much of a mess.

I'd suggest, that `ALIGN()` is unprocessed, because common.h isn't included in memory.h as it really should. But interestingly, adding it there doesn't change the resulting .x, strange.

Anyway, I'd like to just commit this one instead of going your proposed route of reverting and fixing. We have a introduced a harmless regression and it would be good to fix it, if possible for 1.7. After that you can propose any further improvements.